layermanager/layermanager.go

```diff
@@ -227,69 +281,59 @@ func (layermanager *LayerManager) doInstallLayer(
 	if err := os.MkdirAll(extractLayerDir, 0o755); err != nil {
 		return aoserrors.Wrap(err)
 	}
-
 	defer os.RemoveAll(extractLayerDir)
 
-	if err := layermanager.extractPackageByURL(extractLayerDir, layermanager.downloadDir, layerURL, fileInfo); err != nil {
-		return aoserrors.Wrap(err)
-	}
-
-	defer layermanager.deallocatedSpaceExtractDirLayer(extractLayerDir)
-
-	var byteValue []byte
-
-	if byteValue, err = ioutil.ReadFile(path.Join(extractLayerDir, layerOCIDescriptor)); err != nil {
-		return aoserrors.Wrap(err)
-	}
-
-	var layerDescriptor imagespec.Descriptor
-
-	if err = json.Unmarshal(byteValue, &layerDescriptor); err != nil {
+	layerDescriptor, spaceExtract, err := layermanager.extractPackageByURL(extractLayerDir, layerURL, fileInfo)
+	if err != nil {
 		return aoserrors.Wrap(err)
 	}
 
-	var layerPath string
+	defer func() {
+		if err := spaceExtract.Release(); err != nil {
+			log.Errorf("Can't release memory: %v", err)
+		}
+	}()
 
-	if layerPath, err = getValidLayerPath(layerDescriptor, extractLayerDir); err != nil {
+	layerPath, err := getValidLayerPath(layerDescriptor, extractLayerDir)
+	if err != nil {
 		return aoserrors.Wrap(err)
 	}
 
-	if err = layermanager.tryAllocateSpace(layerDescriptor.Size); err != nil {
+	spaceLayer, err := layermanager.layerAllocator.AllocateSpace(uint64(layerDescriptor.Size))
+	if err != nil {
 		return aoserrors.Wrap(err)
 	}
 
-	layerStorageDir := path.Join(layermanager.layersDir,
+	installInfo.Path = path.Join(layermanager.layersDir,
```

Comment:
Here and in other places use filepath.Join instead of path.Join

Reply:
Done